client/scripts/chat.js

```diff
@@ -1,56 +1,146 @@
 export class Chat {
-  constructor (element) {
+  constructor(element) {
     this._chatContainer = element;
     this._messageInput = this._chatContainer.querySelector('#message');
     this._usersList = this._chatContainer.querySelector('#users');
     this._messagesList = this._chatContainer.querySelector('#messages');
     this._chatForm = this._chatContainer.querySelector('#chat-form');
+    this._emojiContainer = this._chatContainer.querySelector('#emoji-container');
+    this._emojiMap = {
+      ':)': 'smile',
+      ':(': 'sad',
+      '(poop)': 'poop',
+      '(heart eyes)': 'hearteyes',
+      '(thumb up)': 'thumbsup'
+    }
   }
 
-  show () {
-    this._chatContainer.classList.remove('hidden');
+  show() {
+    this._imagesMap = {};
+    this._mapEmojiToImages(this._emojiMap)
+        .then((result) => {
+          result.map((item) => {
```

Comment:
It is not what map should be used for, because it should be
a) immutable
b) return new array (array of undefined in your case).

Use either forEach or reduce (preferred)